wp-content/themes/sample-theme/blocks/src/core--button/editor.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * WordPress dependencies
+ */
+import { unregisterBlockStyle } from '@wordpress/blocks';
+
+import './css/editor.scss';
+
+// Unregister default button styles. Note, this can't be done via PHP.
+document.addEventListener( 'DOMContentLoaded', () => {
+    unregisterBlockStyle( 'core/button', 'fill' );
```

Comment:
Turns out PHP `unregister_block_style` doesn't transform and run JS while `register_block_style` does.